election/views.py

```diff
@@ -86,3 +96,52 @@ def create(self, request, *args, **kwargs):
             return Response(status=status.HTTP_400_BAD_REQUEST)
         headers = self.get_success_headers(serializer.data)
         return Response(status=status.HTTP_201_CREATED, headers=headers)
+
+
+
+
+class ResultAPIView(APIView):
+    """ 
+    View to list the result of an election using majority judgment.
+    """
+
+
+    def create(self, request, *args, **kwargs):
+
+        election = request.data.get("election", "")
+
+        try:
+           election = Election.objects.get(id=election)
+        except Election.DoesNotExist:
+            return Response(
+                "unknown election",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        votes = Vote.objects.filter(election=election)
+        scores = mj.votes_to_grades([v.grades_by_candidate for v in votes], \
+                                    election.num_grades)
+        ranks = mj.majority_judgment(scores)
+        
+        class Candidate:
+            def __init__(self, name, rank, votes):
+                self.name = name
+                self.rank = rank
+                self.votes = votes
+
+        class CandidateSerializer(serializers.serializer):
+            name = serializers.CharField(max_length=255)
+            rank = serializers.IntField()
+            scores = serializers.ListField()
+
```

Comment:
What is the rationale of including those classes' definitions IN the controller? Souldn't they belong to the `serializers.py` file (and maybe `models.py`)?

Reply:
I don't like much when very specific code is put somewhere else. There is very unlikely to use these classes somewhere. `models.py` is about database schema, and we don't want to create a table for `Candidate`. Putting only the serializers inside  `serializers.py` looks weird.

Reply:
I globally agree with your view about specific code. However:

- DRF's standards and the current code base place those serializes in a separate files. We should make a consistent choice at least in our code base.
- Define classes INSIDE the handler is not very elegant: the classes are define each time the code is run.

Reply:
ok then I move both classes in `serializers.py` with a comment linking to this thread